docs/usage/library_factories/sqlalchemy_factory.rst

```diff
@@ -36,6 +36,29 @@ By default, this will add generated models to the session and then commit. This
 
 Similarly for ``__async_session__`` and ``create_async``.
 
+SQLAlchemy custom types
+------------------------------
+
+TypeDecorator based
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Type will be inferred from SQLAlchemy type defined in ``impl``.
+
+.. literalinclude:: /examples/library_factories/sqlalchemy_factory/test_example_4.py
+    :caption: Custom types with TypeDecorator
+    :language: python
+
+
+UserDefinedType based
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``get_sqlalchemy_types`` classmethod needs to be overridden to provide factory function for custom type.
```

Comment:
This can be changed so that the user should override the `get_provider_map` method. That makes it more consistent with how the other factories handle custom user defined types.

Reply:
This provides the mapping from sqlalchemy types to provider where this is directly mapped from column vs where the implementation type could be used. 

I think these could be unified but may require some reworking so the the logic of handling types in exposed separately. There wasn't a clear way to map things like `Integer` column without repeating int handling. This case is alright to handle but this becomes more complex especially when considering things like `list[str]`.

Reply:
> This provides the mapping from sqlalchemy types to provider where this is directly mapped from column vs where the implementation type could be used.
> 
> I think these could be unified but may require some reworking so the the logic of handling types in exposed separately. There wasn't a clear way to map things like `Integer` column without repeating int handling. This case is alright to handle but this becomes more complex especially when considering things like `list[str]`.

I agree that we can keep a separate `get_sqlalchemy_types` and a `get_provider_map` so that we can deal with `sql` types in an easier manner. However, since this is a user defined type, should it not go into the `get_provider_map` which is where the user defined types are supposed to be given for the other factories?

Reply:
Not with the current implementation. 

`get_type_from_column` only considers the output of `get_sqlalchemy_types` [here](https://github.com/litestar-org/polyfactory/blob/main/polyfactory/factories/sqlalchemy_factory.py#L117) for SQLA types it should not map for providers. This could be changed to just using `get_provider_map` to bring inline with other factories. 

I don't think this is a backwards breaking change though one very minor concern it may be harder where want to provide a mapping for a field with type of Column itself. This is minor as feels a fairly obscure use case.